packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -316,6 +342,145 @@ export class SharedTree
 	}
 }
 
+/**
+ * Get a {@link BranchableTree} from a {@link ITree}.
+ * @remarks The branch can be used for "version control"-style coordination of edits on the tree.
+ * @alpha
+ */
+export function getBranch(tree: ITree): BranchableTree;
+/**
+ * Get a {@link BranchableTree} from a {@link TreeView}.
+ * @remarks The branch can be used for "version control"-style coordination of edits on the tree.
+ * @alpha
+ */
+export function getBranch(view: TreeView<ImplicitFieldSchema>): BranchableTree;
+export function getBranch(treeOrView: ITree | TreeView<ImplicitFieldSchema>): BranchableTree {
+	if (treeOrView instanceof SharedTree) {
+		return treeOrView;
+	}
+
+	assert(
+		treeOrView instanceof SchematizingSimpleTreeView,
+		"Unsupported TreeView implementation",
+	);
+	return createTreeBranch(treeOrView.checkout, treeOrView.nodeKeyManager, treeOrView.breaker);
+}
+
+export function createTreeBranch(
+	checkout: TreeCheckout,
+	nodeKeyManager: NodeKeyManager,
+	breaker?: Breakable,
+): TreeBranch {
+	return new ForkedTreeBranch(checkout, nodeKeyManager, breaker);
+}
+
+class ForkedTreeBranch implements TreeBranch {
+	private _checkout?: TreeCheckout;
+	private readonly views: TreeView<ImplicitFieldSchema>[] = [];
+
+	public constructor(
+		checkout: TreeCheckout,
+		private readonly nodeKeyManager: NodeKeyManager,
+		private readonly breaker?: Breakable,
+	) {
+		this._checkout = checkout;
+	}
+
+	public get checkout(): TreeCheckout {
+		if (this._checkout === undefined) {
+			throw new UsageError("Branch has been disposed");
```

Comment:
Since this is user-facing, can we expand on the message?

Reply:
Good idea.